src/mrinufft/trajectories/tools.py

```diff
@@ -369,6 +380,290 @@ def unepify(trajectory: NDArray, Ns_readouts: int, Ns_transitions: int) -> NDArr
     return trajectory
 
 
+def _calculate_area(gs, ge, gi, n_down, n_up, n_pl):
+    """Calculate the area traversed by the trapezoidal gradient waveform."""
+    return 0.5 * (gs + gi) * (n_down + 1) + 0.5 * (ge + gi) * (n_up - 1) + n_pl * gi
+
+
+def _calculate_plateau(gs, ge, gi, n_down, n_up, area_needed, ceil=False, buffer=0):
+    """Calculate the plateau length of the trapezoidal gradient waveform."""
+    n_pl = (
+        0.5
+        * (2 * area_needed - gs * (n_down + 1) - ge * (n_up - 1) + gi * (n_down + n_up))
+        / (gi + np.finfo(gi.dtype).eps)
+    )
+    if ceil:
+        n_pl = np.ceil(n_pl).astype(int)
+    return n_pl + buffer
+
+
+def _calculate_ramps(gs, ge, gi, smax, raster_time, ceil=False, buffer=0):
+    """Calculate the number of time steps for the ramp down and up."""
+    n_ramp_down = np.abs(gi - gs) / (smax * raster_time)
+    n_ramp_up = np.abs(ge - gi) / (smax * raster_time)
+    if ceil:
+        n_ramp_down = np.ceil(n_ramp_down).astype(int)
+        n_ramp_up = np.ceil(n_ramp_up).astype(int)
+    return n_ramp_down + buffer, n_ramp_up + buffer
```

Comment:
why the buffer ? Also in the rest of the code it is always equal to 1, consider just inlining it ? (with a comment to say why this buffer is required)